crates/typed-store/src/rocks/mod.rs

```diff
@@ -194,23 +214,44 @@ impl<K, V> DBMap<K, V> {
 pub struct DBBatch {
     rocksdb: Arc<rocksdb::DBWithThreadMode<MultiThreaded>>,
     batch: WriteBatch,
+    db_metrics: Arc<DBMetrics>,
 }
 
 impl DBBatch {
     /// Create a new batch associated with a DB reference.
     ///
     /// Use `open_cf` to get the DB reference or an existing open database.
-    pub fn new(dbref: &Arc<rocksdb::DBWithThreadMode<MultiThreaded>>) -> Self {
+    pub fn new(
+        dbref: &Arc<rocksdb::DBWithThreadMode<MultiThreaded>>,
+        db_metrics: &Arc<DBMetrics>,
+    ) -> Self {
         DBBatch {
             rocksdb: dbref.clone(),
             batch: WriteBatch::default(),
+            db_metrics: db_metrics.clone(),
         }
     }
 
     /// Consume the batch and write its operations to the database
     #[instrument(level = "trace", skip_all, err)]
     pub fn write(self) -> Result<(), TypedStoreError> {
+        let db_name = self
+            .rocksdb
+            .path()
+            .file_name()
+            .and_then(|f| f.to_str())
+            .unwrap_or("unknown");
+        let _timer = self
```

Comment:
The timer here is unnecessary.  We already have automatic span timing measurement for any function that is instrumented with `#[intrument` ... ie each span.  It should be available today.

Reply:
i think i mentioned before, but this tracks metrics by db/column family which is useful and not possible with spans

Reply:
Ok, that's fine, then go ahead.  There is actually a way to add metadata to spans such as the db_name, see 
https://docs.rs/tracing/0.1.36/tracing/attr.instrument.html
however it is not exposed yet to Prometheus as a label.